### PR TITLE
Automate AD app registration

### PR DESCRIPTION
**Pull request by christianacca**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -57,6 +57,16 @@ You also need to add the required API permissions to have '`Azure DevOps user_im
 
 ![aad api permissions](https://imgur.com/aVd51d0.png)
 
+### register-azure-devops-npm-auth.ps1
+
+To automate the creation of the above AD app registration in *your* AD tenant:
+
+1. Ensure you have installed [azure-cli](https://docs.microsoft.com/en-us/cli/azure/install-azure-cli)
+2. Open a powershell core prompt (minimum version 6.1)
```
Comment (yohanb) [inline README.md:65]: does this need to be run as an administrator?

File: register-azure-devops-npm-auth.ps1 [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+        }
+    }
+
+}
+process {
+    try {
+
+        if ($Login) {
+            Write-Information 'Connecting to Azure AD Account...'
+            Invoke-Exe { az login } | Out-Null
```
Comment (yohanb) [inline register-azure-devops-npm-auth.ps1:59]: isn't it better to use the PowerShell Az-Module instead?

**Comment by yohanb** [comment]
Good stuff!  I like you took the time to add the README docs and document your script as well. :+1:

File: register-azure-devops-npm-auth.ps1 (hunk 2 above)
Comment (christianacca) [reply register-azure-devops-npm-auth.ps1:59]: The Az modules are REALLY not complete yet. I found that out the hard way trying to automate app role assignment. So once bitten twice shy!

File: README.md (hunk 1 above)
Comment (christianacca) [reply README.md:65]: yeah the script mentions that you need Global administrator (line 17). This is as per the docs:

image

That, said I think a lesser role might be sufficient (see below), but I hedged my bets to go with the above.

image

Comment (yohanb) [reply README.md:65]: I meant the PowerShell prompt probably needs to be "Run as administrator..." right?

Comment (christianacca) [reply README.md:65]: that a windows thing - I'm on a mac :-)

I'm not sure but I wouldn't expect a windows admin credential (or lack of) would affect the azure login session you acquire via `az login`... but I could well be wrong!

Comment (christianacca) [reply README.md:65]: Yo @yohanb, any more information you need from me?

Comment (yohanb) [reply README.md:65]: @christianacca nope!  All good!  Will merge and publish asap :)

Comment (yohanb) [reply README.md:65]: actually no need to publish the package; PR is merged back!  Thanks for your contribution :+1:

**Pull request merged**